new(usr): Add first pass at Readwise API V3 (support for Reader)

joshbeckman opened the pull request:
Note that this is clearly still in beta (liable to change) because there are cases where the docs don't match the data (e.g. `published_date`).

Addresses https://github.com/joshbeckman/readwise-ruby/issues/1

joshbeckman committed:
new(usr): Add first pass at Readwise API V3 (support for Reader)

Note that this is clearly still in beta (liable to change) because there
are cases where the docs don't match the data (e.g. `published_date`).

Addresses https://github.com/joshbeckman/readwise-ruby/issues/1

joshbeckman committed:
fix(dev): Handle nil tags on document responses

andyw8 commented on lib/readwise/document.rb:
```diff
@@ -0,0 +1,140 @@
+    :parent_id, # both highlights and notes made in Reader are also considered Documents.
+                # Highlights and notes will have `parent_id` set, which is the Document id
+                # of the article/book/etc and highlight that they belong to, respectively.
+    keyword_init: true
```
btw if you can choose to only support Ruby 3.2 or newer then it defaults to `keyword_init`.

joshbeckman replied:
🤷🏼 It's not that burdensome to keep around. Ruby 3.0 isn't EOL until next year, I think.

joshbeckman commented on lib/readwise/client.rb:
```diff
@@ -3,19 +3,58 @@
 require_relative 'book'
 require_relative 'highlight'
 require_relative 'tag'
+require_relative 'document'
```
> [!WARNING]
> tODO: update the README with examples of these methods and a disclaimer that they are in beta and liable to change.

joshbeckman commented on spec/readwise_spec.rb:
```diff
@@ -1,3 +1,5 @@
+require 'readwise'
```
> [!WARNING]
> TODO: Write tests for the new methods

joshbeckman committed:
fix(dev): Handle tags being returned as unstructured strings

joshbeckman committed:
chg(dev): Add debugging dump

joshbeckman committed:
chg(dev): Handle document tags in a Hash

joshbeckman committed:
Add status code to error message

joshbeckman committed:
feat(test): Add tests for document API methods

Ref: https://readwise.io/reader_api

Part of [Add support for Reader API · Issue #1 · joshbeckman/readwise-ruby](https://github.com/joshbeckman/readwise-ruby/issues/1)

joshbeckman committed:
Update README
